plugins/processors/snmp_lookup/lookup.go

```diff
@@ -0,0 +1,206 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package snmp_lookup
+
+import (
+	_ "embed"
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/snmp"
+	si "github.com/influxdata/telegraf/plugins/inputs/snmp"
+	"github.com/influxdata/telegraf/plugins/processors"
+
+	"github.com/gosnmp/gosnmp"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// snmpConnection is an interface which wraps a *gosnmp.GoSNMP object.
+// We interact through an interface so we can mock it out in tests.
+type snmpConnection interface {
+	Host() string
+	Walk(string, gosnmp.WalkFunc) error
+	Get(oids []string) (*gosnmp.SnmpPacket, error)
+	Reconnect() error
+}
+
+type tagMapRows map[string]map[string]string
+type tagMap struct {
+	created time.Time
+	rows    tagMapRows
+}
+
+type Lookup struct {
+	AgentTag string     `toml:"agent_tag"`
+	IndexTag string     `toml:"index_tag"`
+	Tags     []si.Field `toml:"tag"`
+
+	snmp.ClientConfig
+
+	CacheSize             int             `toml:"max_cache_entries"`
+	ParallelLookups       int             `toml:"max_parallel_lookups"`
+	Ordered               bool            `toml:"ordered"`
+	CacheTTL              config.Duration `toml:"cache_ttl"`
+	MinTimeBetweenUpdates config.Duration `toml:"min_time_between_updates"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	translator si.Translator
+	table      si.Table
+
+	cache             *store
+	backlog           *backlog
+	getConnectionFunc func(string) (snmpConnection, error)
+}
+
+const (
+	defaultCacheSize             = 100
+	defaultCacheTTL              = config.Duration(8 * time.Hour)
+	defaultParallelLookups       = 100
```

Comment:
I guess we should reduce the number here to something sensible as 100 threads pinging agents might generate some load. ;-)

Reply:
I was guessing 16 or we use the number of CPUs... But that might be overhead. I would rather start out low and have people increase the setting in the config than blocking out weaker machines...